app/src/userpages/components/Header/Search/search.pcss

```diff
@@ -0,0 +1,110 @@
+.root {
+  align-items: center;
+  cursor: default;
+  display: flex;
+  position: relative;
+  transition: 300ms ease width;
+  width: var(--um);
+}
+
+.root::after {
+  bottom: 0;
+  content: '';
+  display: block;
+  height: 2px;
+  left: 0;
+  position: absolute;
+  transition: 300ms ease background-color;
+  width: 100%;
+}
+
+.open::after {
+  background-color: #FF5C00;
+}
+
+.open {
+  cursor: text;
+  width: calc(12 * var(--um));
+}
+
+.inner {
+  flex-grow: 1;
+  position: relative;
+}
+
+.loupe {
+  color: #A3A3A3;
+  height: var(--um);
+  pointer-events: none;
+  position: absolute;
+  width: var(--um);
+  top: 50%;
+  transform: translateY(-50%);
+  transition: 100ms ease color;
+}
+
+.open .loupe {
+  color: #FF5C00;
+}
+
+.inputWrapper {
+  overflow: hidden;
+  width: 100%;
+}
+
+.root input {
+  border: 0;
+  display: block;
+  height: 2em;
+  line-height: 2em;
+  min-width: calc(2 * var(--um));
+  outline: 0;
+  padding: 0 0 0 calc(2 * var(--um));
+  transform: translateY(-2px);
+  width: 100%;
+}
+
+.root input::placeholder {
+  color: transparent;
+  transition: 300ms linear color;
+}
+
+.open input::placeholder {
+  color: #A3A3A3;
+}
+
+.closeButtonWrapper {
+  flex-shrink: 0;
+  height: calc(0.625 * var(--um));
+  margin-left: calc(0.4375 * var(--um));
+  opacity: 0;
+  overflow: visible;
+  visibility: hidden;
+  transition: 300ms ease;
+  transition-property: visibility, opacity;
+  transition-delay: 300ms, 0s;
+  width: calc(0.625 * var(--um));
+}
+
+.open .closeButtonWrapper {
+  opacity: 1;
+  transition-delay: 0s, 0s;
+  visibility: visible;
+}
+
+.closeButton {
+  appearance: none !important;
+  background-color: transparent;
+  border: 0;
+  display: block;
+  height: calc(1.5 * var(--um));
+  outline: 0 !important;
```

Comment:
would `.root .closeButton` give it enough specificity to avoid invoking `!important`?

Reply:
I was like _noone's watching Mariusz, use `!important`._ Dayum! You're right.